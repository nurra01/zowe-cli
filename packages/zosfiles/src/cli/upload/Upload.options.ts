/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { ICommandOptionDefinition } from "@brightside/imperative";

import i18nTypings from "../-strings-/en";

// Does not use the import in anticipation of some internationalization work to be done later.
const strings = (require("../-strings-/en").default as typeof i18nTypings).UPLOAD.OPTIONS;

/**
 * Object containing all options to be used by the Upload API
 */
export const UploadOptions: {[key: string]: ICommandOptionDefinition} = {
    /**
     * The volume serial
     * @type {ICommandOptionDefinition}
     */
    volume: {
        name: "volume-serial",
        aliases: ["vs"],
        description: strings.VOLUME,
        type: "string"
    },

    /**
     * The binary option
     * @type {ICommandOptionDefinition}
     */
    binary: {
        name: "binary",
        aliases: ["b"],
        description: strings.BINARY,
        type: "boolean"
    },

    /**
     * The migrated recall option
     * @type {ICommandOptionDefinition}
     */
    recall: {
        name: "migrated-recall",
        aliases: ["mr"],
        description: strings.RECALL,
        type: "string",
        defaultValue: "nowait"
    },

    /**
     * The recursive option
     * @type {ICommandOptionDefinition}
     */
    recursive: {
        name: "recursive",
        aliases: ["r"],
        description: strings.RECURSIVE,
        type: "boolean"
    },

    /**
     * The binary_map option
     * @type {ICommandOptionDefinition}
     */
    binary_map: {
        name: "binary_map",
        aliases: ["bm"],
        description: strings.BINARY_MAP,
        type: "string",
        conflictsWith: ["ascii_map"]
    },

    /**
     * The ascii_map option
     * @type {ICommandOptionDefinition}
     */
    ascii_map: {
        name: "ascii_map",
        aliases: ["am"],
        description: strings.ASCII_MAP,
        type: "string",
        conflictsWith: ["binary_map"]
    }
};
