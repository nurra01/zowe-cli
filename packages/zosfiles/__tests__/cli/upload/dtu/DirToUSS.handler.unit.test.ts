/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { Upload } from "../../../../src/api/methods/upload";
import { UNIT_TEST_ZOSMF_PROF_OPTS } from "../../../../../../__tests__/__src__/mocks/ZosmfProfileMock";

describe("Upload dir-to-uss handler", () => {
    describe("process method", () => {
        it("should upload a directory to a USS directory if requested", async () => {
            // Require the handler and create a new instance
            const handlerReq = require("../../../../src/cli/upload/dtu/DirToUssDir.handler");
            const handler = new handlerReq.default();
            const inputdir = "test_dir";
            const USSDir = "USS_dir";

            // Vars populated by the mocked function
            let error;
            let apiMessage = "";
            let jsonObj;
            let logMessage = "";
            let fakeSession = null;

            // Mock the submit JCL function
            Upload.dirToUssDir = jest.fn((session) => {
                fakeSession = session;
                return {
                    success: false,
                    commandResponse: "uploaded",
                    apiResponse: [
                        {success: true, from: inputdir, to: USSDir},
                        {success: false, from: "testfrom", to: "testto"},
                        {success: undefined, from: "dummy", to: "nowhere"}
                    ]
                };
            });

            // Mocked function references
            const profFunc = jest.fn((args) => {
                return {
                    host: "fake",
                    port: "fake",
                    user: "fake",
                    password: "fake",
                    auth: "fake",
                    rejectUnauthorized: "fake",
                };
            });

            try {
                // Invoke the handler with a full set of mocked arguments and response functions
                await handler.process({
                    arguments: {
                        $0: "fake",
                        _: ["fake"],
                        inputdir,
                        USSDir,
                        ...UNIT_TEST_ZOSMF_PROF_OPTS
                    },
                    response: {
                        data: {
                            setMessage: jest.fn((setMsgArgs) => {
                                apiMessage = setMsgArgs;
                            }),
                            setObj: jest.fn((setObjArgs) => {
                                jsonObj = setObjArgs;
                            })
                        },
                        console: {
                            log: jest.fn((logArgs) => {
                                logMessage += "\n" + logArgs;
                            })
                        },
                        progress: {
                            startBar: jest.fn((parms) => {
                                // do nothing
                            }),
                            endBar: jest.fn(() => {
                                // do nothing
                            })
                        }
                    },
                    profiles: {
                        get: profFunc
                    }
                } as any);
            } catch (e) {
                error = e;
            }

            expect(error).toBeDefined();
            expect(profFunc).toHaveBeenCalledWith("zosmf", false);
            expect(Upload.dirToUssDir).toHaveBeenCalledTimes(1);
            expect(Upload.dirToUssDir).toHaveBeenCalledWith(fakeSession, inputdir, USSDir, {
                task: {
                    percentComplete: 0,
                    stageName: 0,
                    statusMessage: "Uploading directory to USS"
                }
            });
            expect(jsonObj).toMatchSnapshot();
            expect(apiMessage).toMatchSnapshot();
            expect(logMessage).toMatchSnapshot();
        });
    });
});
