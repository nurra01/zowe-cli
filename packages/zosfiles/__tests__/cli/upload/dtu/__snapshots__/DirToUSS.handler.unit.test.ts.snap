// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload dir-to-uss handler process method should upload a directory to a USS directory if requested 1`] = `
Object {
  "apiResponse": Array [
    Object {
      "from": "C:\\\\Users\\\\dumal01\\\\Documents\\\\GitHub\\\\WORK\\\\zowe-cli\\\\packages\\\\zosfiles\\\\__tests__\\\\cli\\\\upload\\\\dtu\\\\test_dir",
      "success": true,
      "to": "USS_dir",
    },
    Object {
      "from": "testfrom",
      "success": false,
      "to": "testto",
    },
    Object {
      "from": "dummy",
      "success": undefined,
      "to": "nowhere",
    },
  ],
  "commandResponse": "uploaded",
  "success": false,
}
`;

exports[`Upload dir-to-uss handler process method should upload a directory to a USS directory if requested 2`] = `""`;

exports[`Upload dir-to-uss handler process method should upload a directory to a USS directory if requested 3`] = `
"
[32m- [39m
  [33msuccess: [39m[32mtrue[39m
  [33mfrom: [39m   C:\\\\Users\\\\dumal01\\\\Documents\\\\GitHub\\\\WORK\\\\zowe-cli\\\\packages\\\\zosfiles\\\\__tests__\\\\cli\\\\upload\\\\dtu\\\\test_dir
  [33mto: [39m     USS_dir
[32m- [39m
  [33msuccess: [39m[31mfalse[39m
  [33mfrom: [39m   testfrom
  [33mto: [39m     testto
[32m- [39m
  [33mfrom: [39mdummy
  [33mto: [39m  nowhere


uploaded"
`;
